libs/remix-ui/debugger-ui/src/lib/slider/slider.tsx

```diff
@@ -1,47 +1,42 @@
 import React, { useState, useEffect, useRef } from 'react' // eslint-disable-line
 
 export const Slider = ({ jumpTo, sliderValue, traceLength }) => {
-  const [state, setState] = useState({
-    currentValue: 0
-  })
-
   const onChangeId = useRef(null)
+  const slider = useRef(null)
 
   useEffect(() => {
     setValue(sliderValue)
   }, [sliderValue])
 
   const setValue = (value) => {
-    if (value === state.currentValue) return
-    setState(prevState => {
-      return { ...prevState, currentValue: value }
-    })
-    jumpTo && jumpTo(value)
-  }
-
-  const handleChange = (e) => {
+    if (value === slider.current.value) return
+    slider.current.value = value
     if (onChangeId.current) {
-      window.clearTimeout(onChangeId.current)
+      clearTimeout(onChangeId.current)
     }
     ((value) => {
       onChangeId.current = setTimeout(() => {
-        console.log(value)
-        value = parseInt(value)
-        setValue(value)
+        jumpTo && jumpTo(value)
       }, 100)
-    })(e.target.value)    
+    })(value)
   }
 
+  const handleChange = (e) => {
+    setValue(parseInt(e.target.value))
+  }
+
+  if (slider.current) slider.current.internal_onmouseup = handleChange
```

Comment:
Can you explain this line?

Reply:
Explain?

Reply:
I m using it in the e2e. let me know if there's a better way to handle that.